modules/util/EventEmitterForwarder.js

```diff
@@ -0,0 +1,37 @@
+var EventEmitter = require("events");
+
+/**
+ * Implements utility to forward events from one eventEmitter to another.
+ * @param src {object} instance of EventEmitter or another class that implements
+ * addListener method which will register listener to EventEmitter instance.
+ * @param dest {object} instance of EventEmitter or another class that
+ * implements emit method which will emit an event.
+ */
+function EventEmitterForwarder (src, dest) {
+    if (!src || !dest || typeof(src.addListener) !== "function" ||
+        typeof(dest.emit) !== "function")
+        throw new Error("Invalid arguments passed to EventEmitterForwarder");
+    this.src = src;
+    this.dest = dest;
+}
+
+/**
+ * Adds event to be forwarded from src to dest.
+ * @param srcEvent {string} the event that EventEmitterForwarder is listening
+ * for.
+ * @param dstEvent {string} the event that will be fired from dest.
+ * @param arguments all other passed arguments are going to be fired with
+ * dstEvent.
+ */
+EventEmitterForwarder.prototype.forward = function () {
+    // This line is only for fixing jshint errors.
+    var args = arguments;
+    var srcEvent = args[0];
+    //This will be the "this" value for emit function.
+    args[0] = this.dest;
+    //Using bind.apply to pass the arguments as Array-like object ("arguments")
+    this.src.addListener(srcEvent,
+        this.dest.emit.bind.apply(this.dest.emit, args));
```

Comment:
Wow, that looks like magic even after we've talked about it...

Anyway, let me just ask about the line this.dest.emit.bind.apply(this.dest.emit, args):

I guess the function this.dest.emit that is the first argument (let me call it fun for the sake of clarity) is needed by the bind function as its this. Because that's to be invoked normally as:
````javacript
fun.bind(thisArg, arg1, arg2, ...);
````

But then why do we need this.dest.emit again at the beginning?

I guess part of the confusion for me are all the dots & repetitions. I think you could've gone for some clarity with:
````javascript
var destEmit = this.destEmit;
...destEmit.bind.apply(destEmit, args)...
````

But then have you tried using Function.prototype.bind because that's likely what you want to get to via this.dest.emit.bind at the beginning?
````javascript
Function.prototype.bind.apply(this.dest.emit, args)
````